app/uk/gov/hmrc/gform/gform/DataRetrieveResponseValidator.scala

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.gform
+
+import play.api.libs.json._
+import org.slf4j.{ Logger, LoggerFactory }
+import uk.gov.hmrc.gform.sharedmodel.{ AllowedValueType, AllowedValues, Attr, AttributeInstruction, ConstructAttribute, DataRetrieve, Fetch }
+
+object DataRetrieveResponseValidator {
+
+  private val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  sealed trait ValidationResult
+  case object ValidationSuccess extends ValidationResult
+  case class ValidationFailure(errors: List[String]) extends ValidationResult
```

Comment:
Let's rename this `ValidationResult` to `DataRetrieveValidationResult` and let's define `Success` and `Failure` inside companion object `DataRetrieveValidationResult`